kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/PurchaseOrderAmendmentDocument.java

```diff
@@ -0,0 +1,49 @@
+package edu.arizona.kfs.module.purap.document;
+
+import org.kuali.kfs.sys.businessobject.Building;
+import edu.arizona.kfs.sys.businessobject.BuildingExtension;
+
+public class PurchaseOrderAmendmentDocument extends org.kuali.kfs.module.purap.document.PurchaseOrderAmendmentDocument {
+	
+    protected String routeCode;
+    protected Building buildingObj;
+
+
+    /**
+     * Gets the routeCodeObj attribute. 
+     * @return Returns the routeCodeObj.
+     */
+    public Building getBuildingObj() {
+        return buildingObj;
+    }
+
+    /**
+     * Sets the routeCodeObj attribute value.
+     * @param routeCodeObj The routeCodeObj to set.
+     */
+    public void setBuildingObj(Building buildingObj) {
+        this.buildingObj = buildingObj;
+    }
+    
+    /**
+     * @see org.kuali.kfs.module.purap.document.PurchasingDocument#getRouteCode()
+     */
+    public String getRouteCode() {
+    	
+    	try{
```

Comment:
Should only retrieve route code from Building extension if route code is blank:
if (StringUtils.isBlank(routeCode)) {

try {
BuildingExtension be = (BuildingExtension) buildingObj.getExtension();